game/scripts/vscripts/Sections/Chapter5/Shared.ts

```diff
@@ -57,11 +57,16 @@ export const preRoshKillItems = [
 
 export const allHeroesInfo = friendlyHeroesInfo.concat(enemyHeroesInfo)
 
-function spawnHeroesIfNeeded(location: Vector, heroInfos: HeroInfo[], team: DotaTeam) {
+function spawnHeroesIfNeeded(location: Vector, heroInfos: HeroInfo[], team: DotaTeam, pacifists: boolean = false) {
     const steps = (context: tg.TutorialContext) => heroInfos.map(heroInfo => {
         if (!unitIsValidAndAlive(context[heroInfo.name])) {
             return tg.seq([
                 tg.spawnUnit(heroInfo.name, location.__add(RandomVector(200)), team, heroInfo.name, true),
+                tg.immediate((ctx) => {
+                    if (pacifists) {
+                        return setUnitPacifist(ctx[heroInfo.name], true)
```

Comment:
I don't think setUnitPacifists returns anything?

Reply:
It doesn't, fixed now